airflow/providers/amazon/aws/sensors/redshift.py

```diff
@@ -37,18 +37,25 @@ def __init__(
         self,
         *,
         cluster_identifier: str,
-        target_status: str = 'available',
+        target_status: RedshiftClusterStates = RedshiftClusterStates.AVAILABLE,
         aws_conn_id: str = 'aws_default',
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.cluster_identifier = cluster_identifier
-        self.target_status = target_status
+        self.target_status = (
+            target_status
+            if isinstance(target_status, RedshiftClusterStates)
+            else RedshiftClusterStates(str(target_status))
+        )
+
         self.aws_conn_id = aws_conn_id
         self.hook: Optional[RedshiftHook] = None
 
     def poke(self, context):
-        self.log.info('Poking for status : %s\nfor cluster %s', self.target_status, self.cluster_identifier)
+        self.log.info(
+            'Poking for status : %s\nfor cluster %s', self.target_status.value, self.cluster_identifier
```

Comment:
```suggestion
            'Checking cluster %r for status %r', self.cluster_identifier, self.target_status.value
```
or
```suggestion
            'Checking cluster %r\nwaiting for status %r', self.cluster_identifier, self.target_status.value
```

because `for cluster %s` is not really a grammatical expression by itself and it's on a newline so it seems that it should be.